pkg/sql/logictest/testdata/logic_test/drop_owned_by

```diff
@@ -0,0 +1,318 @@
+# database privileges. However, the root user should be able to drop testuser's
+# database privileges via DROP OWNED BY.
+subtest revoke-privileges-db
+
+query TTTB
+SHOW GRANTS ON DATABASE test
+----
+test  admin     ALL      true
+test  public    CONNECT  false
+test  root      ALL      true
+test  testuser  ALL      true
+
+user root
+
+statement ok
+DROP OWNED BY testuser
+
+query TTTB
+SHOW GRANTS ON DATABASE test
+----
+test  admin   ALL      true
+test  public  CONNECT  false
+test  root    ALL      true
+
+# REVOKE-PRIVILEGES-SCHEMA: Test that DROP OWNED BY revokes privileges on
+# schemas in the current database.
+#
+# In this test, root creates a schema and grants privileges for the schema to
+# testuser. When testuser issues a DROP OWNED BY, those privileges should be
+# revoked.
+subtest revoke-privileges-schema
+
+user root
+
+statement ok
+CREATE SCHEMA s
+
+statement ok
+GRANT CREATE ON SCHEMA s TO testuser WITH GRANT OPTION
+
+user testuser
+
+statement ok
+CREATE TABLE s.t()
+
+statement ok
+DROP OWNED BY testuser
+
+query TTTTB
+SHOW GRANTS ON SCHEMA s
+----
+test  s  admin  ALL  true
+test  s  root   ALL  true
+
+query TTTTIT
+SHOW TABLES FROM s
+----
+
+user root
+
+statement ok
+DROP SCHEMA s
+
+# REVOKE-PRIVILEGES-TABLE: Test that DROP OWNED BY revokes privileges on
+# objects in the database.
+subtest revoke-privileges-table
+
+user root
+
+statement ok
+CREATE TABLE t()
+
+statement ok
+GRANT ALL ON t TO testuser WITH GRANT OPTION
+
+user testuser
+
+query TTTTTB
+SHOW GRANTS ON t
+----
+test  public  t  admin    ALL  true
+test  public  t  root     ALL  true
+test  public  t  testuser ALL  true
+
+statement ok
+DROP OWNED BY testuser
+
+query TTTTTB
+SHOW GRANTS ON t
+----
+test  public  t  admin    ALL  true
+test  public  t  root     ALL  true
+
+user root
+
+statement ok
+DROP TABLE t
+
+# MUTIROLE: Test DROP OWNED BY with multiple roles.
+subtest multirole
+
+statement ok
+CREATE ROLE r1
+
+statement ok
+CREATE ROLE r2
+
+statement ok
+SET ROLE r1
+
+statement ok
+CREATE TABLE t1()
+
+statement ok
+SET ROLE r2
+
+statement ok
+CREATE TABLE t2()
+
+statement ok
+SET ROLE root
+
+query TTTTIT
+SHOW TABLES FROM public
+----
+public  t1  table  r1   0  NULL
+public  t2  table  r2   0  NULL
+
+statement ok
+DROP OWNED BY r1, r2
+
+query TTTTIT
+SHOW TABLES FROM public
+----
+
+# ROLES: Test that the current user is a member of all the specified roles. The
+# admin role and the root user are off-limits.
+subtest roles
+
+user testuser
+
+statement error pq: permission denied to drop objects
+DROP OWNED BY testuser2
+
+statement error pq: permission denied to drop objects
+DROP OWNED BY testuser, testuser2
+
+statement error pq: cannot drop objects owned by role "root" because they are required by the database system
+DROP OWNED BY root
+
+statement error pq: cannot drop objects owned by role "admin" because they are required by the database system
+DROP OWNED BY admin
```

Comment:
I changed the unimplemented error from the one in `scerror` to the one in `unimplemented`, since it seems like the former causes us to try to fall back to the old schema changer, which ends up throwing a less descriptive error message for `DROP OWNED BY`. I added the logictest check but not the `unimplemented_drop` one--is this the right call?

Reply:
Yeah, I hear you. I'm good with this approach since we're going to implement it anyway. But I'm wondering if you could just set new schema changer's mode to `unsafe_always` since you disable the legacy schema changer configuration in this logic test :) also cc @postamar if you have strong opinion on the options here.

Reply:
Yes, absolutely. Thanks.